code/modules/atmospherics/machinery/components/binary_devices/binary_atmos_base.dm

```diff
@@ -153,3 +153,9 @@
 /obj/machinery/atmospherics/binary/process_atmos()
 	..()
 	return parent1 && parent2
+
+/obj/machinery/atmospherics/binary/relaymove(mob/living/user, direction)
+	var/obj/machinery/atmospherics/target_move = findConnecting(direction)
+	..()
+	user.remove_ventcrawl() //we moved between pipe networks, no more seeing into the old one
+	user.add_ventcrawl(src, target_move)
```

Comment:
Yeah this is gonna recompute the ventcrawl image with every single move, which is going to be very expensive and could possibly become a lag vector.

Reply:
It's only called when moving through binary devies (pumps / valves / uniersal adaptors), not regular pipes.

Reply:
only these devices will recompute the ventcrawl images when crawled through 
![image](https://user-images.githubusercontent.com/16618648/176760607-20ba5c4d-7024-4f6c-b154-c72786578935.png)
universal adapters weren't the problem, it was these devices (binary atmos devices)

Reply:
Its just as costly as starting a vent crawl and it only occurs when moving through these devices. Moving through trinary devices also had this problem, which I've added a fix to in the latest commit.